### PR TITLE
Plot api

### PR DESCRIPTION
**Pull request by danebell**
There are two changes here:
1. Improvements to the `term-freq` endpoint: additional arguments provide (a) a second, conditioning variable to allow two-way counting, and (b) a regular expression string to filter the terms returned.
2. A new endpoint, `term-hist` which returns a frequency table of terms binned either by user specification or by a sensible default dependent on whether the bins should be fixed-width or proportional.

Tests are included.

### CODE REVIEW

File: backend/app/controllers/OdinsonController.scala [hunk 1]
```diff
@@ -93,6 +97,20 @@ class OdinsonController @Inject() (config: Config = ConfigFactory.load(), cc: Co
     pretty: Option[Boolean]
   ) = Action.async {
     Future {
+
+      // if a regex filter is specified, apply it
+      def isMatch(s: String, filter: Option[String]): Boolean = {
```
Comment (myedibleenso) [inline backend/app/controllers/OdinsonController.scala:102]: Let's add a docstring to this method.

File: backend/test/controllers/OdinsonControllerSpec.scala [hunk 2]
```diff
@@ -183,6 +183,17 @@ class OdinsonControllerSpec extends PlaySpec with GuiceOneAppPerTest with Inject
       println(Helpers.contentAsString(response))
     }
 
+    "respond with frequency table using the /term-hist endpoint" in {
+      val body = Json.obj("field" -> "lemma")
+
+      val response =
+        route(app, FakeRequest(GET, "/api/term-hist?field=lemma").withJsonBody(body)).get
```
Comment (myedibleenso) [inline backend/test/controllers/OdinsonControllerSpec.scala:190]: Can we add another test that includes some of the other optional params for this endpoint (ex. log scale)?

**Review by myedibleenso — CHANGES_REQUESTED** [review]

**Comment by danebell** [comment]
I have made the requested changes, and added two endpoints `rule-freq` and `rule-hist` to provide analogous statistical frequency tables to `term-freq` and `term-hist`, respectively.

**Comment by myedibleenso** [comment]
Thank you, @danebell !

**Pull request merged**